engine/common/follower/pending_tree/pending_tree.go

```diff
@@ -147,28 +141,38 @@ func (t *PendingTree) AddBlocks(certifiedBlocks []CertifiedBlock) ([]CertifiedBl
 func (t *PendingTree) connectsToFinalizedBlock(block CertifiedBlock) bool {
 	if block.Block.Header.ParentID == t.lastFinalizedID {
 		return true
-	} else if parentVertex, found := t.forest.GetVertex(block.Block.Header.ParentID); found {
+	}
+	if parentVertex, found := t.forest.GetVertex(block.Block.Header.ParentID); found {
 		return parentVertex.(*PendingBlockVertex).connectedToFinalized
-	} else {
-		return false
 	}
+	return false
 }
 
 // FinalizeForkAtLevel takes last finalized block and prunes levels below the finalized view.
 // When a block is finalized we don't care for all blocks below it since they were already finalized.
+// Finalizing a block might result in observing a connected chain of blocks that previously weren't.
+// These blocks will be returned as result of invocation.
 // No errors are expected during normal operation.
-func (t *PendingTree) FinalizeForkAtLevel(finalized *flow.Header) error {
+func (t *PendingTree) FinalizeForkAtLevel(finalized *flow.Header) ([]CertifiedBlock, error) {
```

Comment:
What does the "AtLevel" part of the method name mean? We could just write:
```suggestion
func (t *PendingTree) FinalizeFork(finalizedHead *flow.Header) ([]CertifiedBlock, error) {
```